apps/fabric-website/src/components/IconGrid/IconGrid.tsx

```diff
@@ -56,36 +58,84 @@ export class IconGrid extends React.Component<IIconGridProps, IIconGridState> {
+    switch (this.props.iconType) {
+      case 'font':
+        // The explicit json-loader! import helps prevent this from being processed by TS
+        // (we want only webpack to handle it)
+        const fontIcons = (await import('json-loader!@uifabric/icons/lib/data/AllIconNames.json')) as IIconInfo[];
+        // There's also a metadata entry in this file which doesn't have a name
+        this.setState({ icons: fontIcons.filter(icon => !!icon.name) });
+        break;
+
+      case 'core':
+        const coreIcons = (await import('json-loader!office-ui-fabric-core/src/data/icons.json')) as IIconInfo[];
+        this.setState({ icons: coreIcons });
+        break;
+
+      case 'svg':
+        const MDL2Icons = await import('@fluentui/react-icons');
+        this.setState({
+          icons: Object.keys(MDL2Icons)
+            .map((name): IIconInfo | undefined => {
+              const IconComponent = MDL2Icons[name];
+              if (typeof IconComponent === 'function' && /^[A-Z].*Icon$/.test(name)) {
+                return { name, icon: IconComponent };
+              }
+              return undefined;
+            })
+            .filter(icon => !!icon),
+        });
+        break;
+    }
+  }
+
   private _getItems = (): { name: string }[] => {
-    const { icons } = this.props;
-    const { searchQuery } = this.state;
+    const { icons, searchQuery } = this.state;
+    const query = searchQuery.toLowerCase();
 
-    return icons.filter(icon => icon && icon.name && icon.name.toLowerCase().indexOf(searchQuery.toLowerCase()) !== -1);
+    return icons ? (query ? icons.filter(icon => icon.name?.toLowerCase().indexOf(query) !== -1) : icons) : [];
   };
 
-  private _renderIcon = (icon: { name: string }, index?: number): JSX.Element => {
-    const { useFabricIcons } = this.props;
+  private _renderIcon = (icon: IIconInfo): JSX.Element => {
+    const { iconType } = this.props;
     let iconClassName = `ms-Icon ms-Icon--${icon.name}`;
-    const iconRef = this._iconRefs[icon.name];
-    if (iconRef.current && iconRef.current.offsetWidth > 80) {
-      iconClassName += ' hoverIcon';
-    }
 
-    return (
-      <li key={icon.name + index} aria-label={icon.name + ' icon'}>
-        {useFabricIcons ? (
-          <Icon iconName={icon.name} />
-        ) : (
-          <i ref={iconRef} className={iconClassName} title={icon.name} aria-hidden="true" />
-        )}
-        <span>{icon.name}</span>
-      </li>
-    );
+    switch (iconType) {
+      case 'font':
+        return (
+          <li key={icon.name} title={icon.name} aria-label={icon.name + ' icon'}>
+            <Icon iconName={icon.name} />
+          </li>
+        );
+      case 'svg':
+        const IconComponent = icon.icon;
+        return (
+          <li key={icon.name} title={icon.name} aria-label={icon.name + ' icon'}>
+            <IconComponent />
+          </li>
+        );
+      case 'core':
+        if (!this._iconRefs[icon.name]) {
```

Comment:
Changed this to only create the refs for the icons where they're being used. I also have no idea what's up with this styling logic in general, but seemed safest to leave it as-is.